src/data/bucket/line_bucket.js

```diff
@@ -50,6 +50,8 @@ const lineInterface = {
         {property: 'line-opacity', multiplier: 10, type: 'Uint8'},
         {property: 'line-gap-width', multiplier: 10, type: 'Uint8', name: 'a_gapwidth'},
         {property: 'line-offset', multiplier: 1, type: 'Int8'},
+        {property: 'line-width', multiplier: 10, type: 'Uint8', name: 'a_width'},
+        {property: 'line-width', multiplier: 10, type: 'Uint8', name: 'a_floorwidth'},
```

Comment:
This approach looks great!

I wonder if adding a `floorZoom: true` to the property definition here and using that to making the `if (name === 'a_floorwidth')` exceptions less exceptional would be good. Or if that would just hide the exception in a bad way.